### PR TITLE
Hierarchy level size limiting support

### PR DESCRIPTION
**Pull request by grigasp**
Part of https://github.com/iTwin/itwinjs-backlog/issues/104

Prerequisites:
- https://github.com/iTwin/imodel-native/pull/187
- https://github.com/iTwin/itwinjs-core/pull/5168
- [x] The above changes released
- [x] Update `itwinjs-core` in `devDependencies` to consume the above changes

Follow-up items:
- https://github.com/iTwin/presentation/issues/106
- https://github.com/iTwin/presentation/issues/88

### CODE REVIEW

File: packages/components/src/test/instance-filter-builder/InstanceFilterConverter.test.ts [hunk 1]
```diff
@@ -367,9 +367,9 @@ describe("convertToInstanceFilterDefinition", () => {
   });
 
   describe("returns base properties class", () => {
-    async function* asyncGenerator(): AsyncIterableIterator<any> {
-      return;
-    }
+    // async function* asyncGenerator(): AsyncIterableIterator<any> {
```
Comment (saskliutas) [inline packages/components/src/test/instance-filter-builder/InstanceFilterConverter.test.ts:370]: Should be removed?

File: packages/components/src/test/instance-filter-builder/PresentationInstanceFilterBuilder.test.tsx [hunk 2]
```diff
@@ -83,10 +83,10 @@ describe("PresentationInstanceFilter", () => {
     await Presentation.initialize();
     Element.prototype.scrollIntoView = sinon.stub();
 
-    async function* generator() {
-      return;
-    }
-    imodelMock.setup((x) => x.query(moq.It.isAnyString(), moq.It.isAny(), moq.It.isAny())).returns(() => generator());
+    // async function* generator() {
```
Comment (saskliutas) [inline packages/components/src/test/instance-filter-builder/PresentationInstanceFilterBuilder.test.tsx:86]: Should be removed?

File: packages/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx [hunk 3]
```diff
@@ -62,10 +62,10 @@ describe("PresentationInstanceFilterDialog", () => {
     await UiComponents.initialize(localization);
     await Presentation.initialize();
 
-    async function* generator() {
-      return;
-    }
-    imodelMock.setup((x) => x.query(moq.It.isAnyString(), moq.It.isAny(), moq.It.isAny())).returns(() => generator());
+    // async function* generator() {
```
Comment (saskliutas) [inline packages/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx:65]: Should be removed?

**Review by saskliutas — APPROVED** [review]

File: packages/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx (hunk 3 above)
Comment (grigasp) [reply packages/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx]: yes, cleaned up

File: packages/components/src/test/instance-filter-builder/PresentationInstanceFilterBuilder.test.tsx (hunk 2 above)
Comment (grigasp) [reply packages/components/src/test/instance-filter-builder/PresentationInstanceFilterBuilder.test.tsx]: yes, cleaned up

File: packages/components/src/test/instance-filter-builder/InstanceFilterConverter.test.ts (hunk 1 above)
Comment (grigasp) [reply packages/components/src/test/instance-filter-builder/InstanceFilterConverter.test.ts]: yes, cleaned up

**Comment by ben-polinsky** [comment]
> @ben-polinsky, the docs build now fails with:
> 
> ```
>  Topic ECMetadataProvider does not have a @docs-group-description in presentation-components.ts. Please add a description.
>  Topic MultiTagSelect does not have a @docs-group-description in presentation-components.ts. Please add a description.
>  Topic NavigationPropertyTargetSelector does not have a @docs-group-description in presentation-components.ts. Please add a description.
>  Topic TreeReloader does not have a @docs-group-description in presentation-components.ts. Please add a description.
>  Topic UseColumns does not have a @docs-group-description in presentation-components.ts. Please add a description.
>  Topic UseNavigationPropertyTargetsLoader does not have a @docs-group-description in presentation-components.ts. Please add a description.
>  Topic UseRows does not have a @docs-group-description in presentation-components.ts. Please add a description.
>  Topic UseTableOptions does not have a @docs-group-description in presentation-components.ts. Please add a description.
> ```
> 
> All of these are `@internal` APIs. Are we now required to supply docs for internal APIs too, or is this a bug in BeMetalsmith?

This is part of the docs build I'm least familiar with. Nothing has changed with regards to this behavior, though. I can look into it.

**Pull request merged**